stacker/lookups/handlers/dynamodb.py

```diff
@@ -0,0 +1,129 @@
+from botocore.exceptions import ClientError
+import re
+from stacker.session_cache import get_session
+
+from ...util import read_value_from_path
+
+TYPE_NAME = "dynamodb"
+
+
+def handler(value, **kwargs):
+    """Get a value from a dynamodb table
+
+    dynamodb field types should be in the following format:
+
+        [<region>:<tablename>@]<primarypartionkey>.<keyvalue>.<keyvalue>...
+
+    Note: The region is optional, and defaults to the environment's
+    `AWS_DEFAULT_REGION` if not specified.
+
+
+    """
+    value = read_value_from_path(value)
+    table_info = None
+    table_keys = None
+    region = None
+    table_name = None
+
+    if "@" in value:
+        table_info, table_keys = value.split("@", 1)
+        if ":" in table_info:
+            region, table_name = table_info.split(":", 1)
+        else:
+            table_name = table_info
+    else:
+        raise ValueError('Please make sure to include a tablename')
+
+    if table_name is None:
+        raise ValueError('Please make sure to include a dynamodb table name')
+
+    table_lookup, table_keys = table_keys.split(":", 1)
+    table_keys = table_keys.split(".")
+
+    clean_table_keys = []
+    new_keys = []
+    regex_matcher = "\[([^\]]+)]"
+
+    # we need to parse the key lookup passed in
+    for key in table_keys:
+        match = re.search(regex_matcher, key)
+        if match:
+            if match.group(1) in ["M", "S", "N", "L"]:
+                match_val = str(match.group(1))
+                key = key.replace(match.group(0), "")
+                new_keys.append({match_val: key})
+                clean_table_keys.append(key)
+            else:
+                raise ValueError(
+                    'Stacker does not support looking up that datatype')
+        else:
+            new_keys.append({"S": key})
+            clean_table_keys.append(key)
+
+    projection_expression = _buildProjectionExpression(clean_table_keys)
+
+    # lookup the data from dynamodb
+    dynamodb = get_session(region).client('dynamodb')
+    try:
+        response = dynamodb.get_item(
+            TableName=table_name,
+            Key={
+                table_lookup: new_keys[0]
+            },
+            ProjectionExpression=projection_expression
+        )
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'ResourceNotFoundException':
+            raise ValueError('Cannot find the dynamodb table: ' + table_name)
+        elif e.response['Error']['Code'] == 'ValidationException':
+            raise ValueError('No dynamo record matched that partition key')
```

Comment:
You probably want a generic `raise` by itself at the end here.  If there's a ClientError, and it doesn't match the two you have here, we almost surely want to raise it.